src/mpid/common/genq/mpidu_genq_shmem_pool.h

```diff
@@ -13,14 +13,10 @@
 #include <stdint.h>
 #include <string.h>
 
-typedef void *MPIDU_genq_shmem_pool_t;
```

Comment:
This transparent pool handle type should be in `mpidu_genq_shmem_pool.h`. The user should be not be looking at the internal `mpidu_genqi_shmem_types.h` for it.

Reply:
User are not accessing the fields of `shmem_pool` nor `shmem_queue`. Both of them are opaque to user. I guess we are arguing for compiler enforcement, which I think it is a minor issue and not worth extra code just for the sake of it.

Reply:
The meaning of this is that `MPIDU_genq_shmem_pool_t pool` becomes an input parameter semantically.

Reply:
I see. Will fix.

Reply:
Both pool and queue have been fully abstracted. Queue no longer depends on pool.